nixos/modules/system/etc/setup-etc.pl

```diff
@@ -30,23 +32,15 @@ sub atomic_symlink {
 # Returns 1 if the argument points to the files in /etc/static.  That
 # means either argument is a symlink to a file in /etc/static or a
 # directory with all children being static.
-sub is_static {
-    my $path = shift;
-
+sub is_static ($path) {
     if (-l $path) {
         my $target = readlink($path);
         return substr($target, 0, length($static)) eq $static;
     }
 
     if (-d $path) {
-        opendir(my $dh, $path) or return 0;
-        my @names = readdir($dh) or die("Failed reading directory `$dh`");
-        closedir($dh);
-
+        my @names = read_dir($path) or die("Failed reading directory `$path`");
```

Comment:
```suggestion
        my @names = read_dir($path);
```

`read_dir()`'s default failure mode is to `croak`, which triggers a fatal error. So the `die` will here only trigger if the directory is empty.